Screens/Signup.js

```diff
@@ -1,75 +1,132 @@
+        businessName: "",
+        errorMessage: null,
+    };
 
-	handleSignUp = () => {
-		firebase
-			.auth()
-			.createUserWithEmailAndPassword(this.state.email, this.state.password)
-			.then(() => this.props.navigation.navigate("Main"))
-			.catch(error => this.setState({ errorMessage: error.message }));
-	};
-	render() {
-		return (
-			<View style={styles.container}>
-				<Text>Sign Up</Text>
-				{this.state.errorMessage && (
-					<Text style={{ color: "red" }}>{this.state.errorMessage}</Text>
-				)}
-				<TextInput
-					placeholder="Email"
-					autoCapitalize="none"
-					style={styles.textInput}
-					onChangeText={email => this.setState({ email })}
-					value={this.state.email}
-				/>
-				<TextInput
-					secureTextEntry
-					placeholder="Password"
-					autoCapitalize="none"
-					style={styles.textInput}
-					onChangeText={password => this.setState({ password })}
-					value={this.state.password}
-				/>
-				<Button title="Sign Up" onPress={this.handleSignUp} />
-				<Button
-					title="Already have an account? Login"
-					onPress={() => this.props.navigation.navigate("Login")}
-				/>
-				<Button
-					title="Login via OTP"
-					onPress={() => this.props.navigation.navigate("Otp")}
-				/>
-			</View>
-		);
-	}
+    componentDidMount() {
+        //1.
+        // const value=AsyncStorage.gettItem("key");
+        //if (value !== null){}
+        //2.
+        // firebase.auth().onAuthStateChanged(user => {
+        //   if (user != null) {
+        //     console.log("We are authenticated now!");
+        //   }
+        // });
+    }
+
+    handleSignUp = () => {
+        // Check if business switch is toggled and no business name is given.
+        // Return error if that is the case
+        if (this.state.isBusiness == true && this.state.businessName === "") {
+            this.setState({ errorMessage: "Business name missing" });
+        } else {
+            // create a user with email and password input
+            firebase
+                .auth()
+                .createUserWithEmailAndPassword(
+                    this.state.email,
+                    this.state.password
+                )
+                .then(() => this.props.navigation.navigate("Login"))
+                .catch((error) =>
+                    this.setState({ errorMessage: error.message })
+                );
+            // If registering as a business add to firestore database
+            if (this.state.isBusiness && this.state.BusinessName != "") {
+                firebase.firestore().collection("businesses").add({
+                    UID: firebase.auth().currentUser.uid,
+                    BusinessName: this.state.businessName,
+                });
+            }
+        }
+    };
+    render() {
+        //define text box for business name input for conditional rendering
+        const businessBox = (
+            <TextInput
+                placeholder="Business Name"
+                autoCapitalize="none"
```

Comment:
Please add a GeoLocation for businesses as a field. What I mean is add another TextInput asking for user location. We will need to store business location and their radius so we know how far a detailer can travel. We can do this next iteration or this one your choice.

Reply:
Currently, location is automatically retrieved on component mount
Only able to take in a properly formatted firestore geopoint at the moment (auto filled once permission is given)

TODO
enable ability to enter an address and convert to lat, long -> fs.geopoint
also allow business to define range willing to travel from location